clang-tidy: Enable more warnings

daandemeyer opened the pull request.

daandemeyer committed:
clang-tidy: Enable more warnings

daandemeyer committed:
sd-journal: Fix return value coding style

Let's always use ret, ret_data and ret_size for output arguments.

size is better than length in this case because we're dealing with
arbitrary, possibly binary, data and not strings.

keszybz reviewed (approved):
LGTM.

daandemeyer committed:
tree-wide: Various coding style cleanups

daandemeyer committed:
clang-tidy: Block system headers with errors

blkid.h and gmessages.h both use const for arguments that are passed
by value, which is pointless and triggers clang-tidy warnings, so exclude
them from processing.

YHNdnzj commented on src/basic/string-table.h:
```diff
@@ -30,45 +29,44 @@ ssize_t string_table_lookup_from_string_fallback(const char * const *table, size
 
-#define _DEFINE_STRING_TABLE_LOOKUP(name, type, scope)                    \
-        _DEFINE_STRING_TABLE_LOOKUP_TO_STRING(name, type, scope)          \
+#define _DEFINE_STRING_TABLE_LOOKUP(name, type, arg, scope)                    \
```
Frankly, I'd much more prefer if we just declare the prototypes in a macro too, and name the params consistently s/i. That also gives you one extra benefit that pure and const attributes can be implied everywhere

yuwata replied:
Yeah, I totally agree with the suggestion. We already have CONFIG_PARSER_PROTOTYPE().

keszybz commented on .clang-tidy:
```diff
@@ -28,7 +28,9 @@ CheckOptions:
 '
+  misc-header-include-cycle.IgnoredFilesList: 'glib-2.0'
 WarningsAsErrors: '*'
+ExcludeHeaderFilterRegex: 'blkid\.h|gmessages\.h'
```
For the commit message:
> const for arguments that are passed by value, which is pointless and triggers clang-tidy warnings

The second justification clause is true. The first one is false. It is not pointless to define something as const. It tells the reader (and the compiler) that the variable is not meant to be modified in the function. Languages more modern than C make extensive use of such annotations. E.g. Java uses `const` a lot. Rust in fact inverts the logic, and requires *mutable* variables to be annotated appropriately, based on the premise that most local variables do not need to be mutable, and code which does mutate local variables is easier to understand.

keszybz commented on src/basic/process-util.c:
```diff
@@ -2304,7 +2304,7 @@ static const char *const sigchld_code_table[] = {
 };
 
-DEFINE_STRING_TABLE_LOOKUP(sigchld_code, int);
+DEFINE_STRING_TABLE_LOOKUP(sigchld_code, int, i);
```
Yeah, this does not make the code more readable. It's just another case where we make the code harder to read to allow a helper tool which doesn't work great to be used.

keszybz commented on src/analyze/analyze-compare-versions.c:
```diff
@@ -33,19 +33,20 @@ int verb_compare_versions(int argc, char *argv[], void *userdata) {
                  * We don't use named values because 11 and 12 don't have names. */
                 return r < 0 ? 12 : r > 0 ? 11 : 0;
 
-        } else {
```
For the commit message: you again use "pointless" without justification. It'd be better to say "not required" or something like that. One good reason to use `else` in such cases is to make the code easier to read by giving it a symmetrical structure. In other cases, it was used mainly to save a line in output by avoiding a line with a closing brace. Either way, not "pointless".

Looking at the whole patch, this introduces an insane amount of churn to programatically enforce a rule that is at best a style guideline. I'd strongly prefer to drop this patch. It doesn't make the code better, and it'll make our development process harder, requiring human writers to jump through hoops in cases where they want to use the `else` in a way that the stupid automatical checker doesn't understand. Please drop this one.

daandemeyer replied:
I mean style guidelines are there to be followed no? I don't think the code is noticeably less readable and it means we never have to comment "please drop the else" on PR reviews again.

If others think the same I'm happy to drop the commit though.

keszybz replied:
I still think that the benefit here is very small. In many cases the code is more readable after the patch, but in many cases it's not. And I think that the churn is has a large cost and overall, I think this patch has negative value.

daandemeyer committed:
systemctl-daemon-reload: Add missing parameter name

daandemeyer committed:
sd-json: Fix sd_json_variant_type_to_string parameter name

The definition will use i because of the macro, so
let's use i in the declaration as well. We can't
use DECLARE_STRING_TABLE_LOOKUP_TO_STRING() because
sd-json.h is a libsystemd public header.

daandemeyer committed:
sd-netlink: Use same argument name for rtnl definitions

We already have the same argument name as used in the
declarations available, let's use it so the declarations
and definitions match.

daandemeyer committed:
libudev: Fix declaration argument names

Make sure these match the definition argument names.

daandemeyer committed:
basic: Gate conditional includes behind ifdef

daandemeyer committed:
locale: Gate conditional includes behind ifdef

Let's make sure clang-tidy doesn't complain on
distributions without locale-gen.

daandemeyer committed:
linter: Remove unneeded meson options

Now that clang-tidy passes regardless of whether
these options are enabled or not, let's stop
configuring them explicitly

yuwata commented on src/systemd/sd-journal.h:
```diff
@@ -120,20 +120,20 @@ int sd_journal_seek_monotonic_usec(sd_journal *j, sd_id128_t boot_id, uint64_t u
 
 int sd_journal_get_cutoff_realtime_usec(sd_journal *j, uint64_t *from, uint64_t *to);
-int sd_journal_get_cutoff_monotonic_usec(sd_journal *j, const sd_id128_t boot_id, uint64_t *from, uint64_t *to);
+int sd_journal_get_cutoff_monotonic_usec(sd_journal *j, sd_id128_t boot_id, uint64_t *from, uint64_t *to);
```
The change should be moved to another commit.

Or, mention the change in the commit message.

keszybz replied:
So we have a separate commit with a description, which is nice. Nevertheless, the description we have is bogus. As discussed previously, the const is not pointless.

yuwata commented on src/basic/proc-cmdline.c:
```diff
@@ -147,15 +147,14 @@ static int proc_cmdline_strv_internal(char ***ret, bool filter_pid1_args) {
                 *ret = TAKE_PTR(args);
                 return 0;
 
```
Please drop the empty line.

yuwata commented on src/basic/process-util.c:
```diff
@@ -898,8 +898,8 @@ int pidref_wait_for_terminate_and_check(const char *name, PidRef *pidref, WaitFl
 
                 return status.si_status;
 
```
Also here. Please drop the empty line.

yuwata commented on src/basic/socket-util.c:
```diff
@@ -1491,13 +1491,13 @@ int sockaddr_un_set_path(struct sockaddr_un *ret, const char *path) {
                 memcpy(ret->sun_path + 1, path + 1, l); /* copy *with* trailing NUL byte */
                 return (int) (offsetof(struct sockaddr_un, sun_path) + l); /* 🔥 *don't* 🔥 include trailing NUL in size */
 
```
Also here.

yuwata commented on src/basic/string-util.c:
```diff
@@ -258,7 +258,9 @@ static size_t ansi_sequence_length(const char *s, size_t len) {
                         return i + 1;
                 return 0;  /* Bad sequence */
 
```
Also here.

yuwata commented on src/basic/time-util.c:
```diff
@@ -96,14 +96,15 @@ usec_t map_clock_usec_raw(usec_t from, usec_t from_base, usec_t to_base) {
 
                 return to_base + delta;
 
```
Also here.

yuwata commented on src/core/cgroup.c:
```diff
@@ -1054,26 +1054,24 @@ uint64_t cgroup_context_cpu_weight(CGroupContext *c, ManagerState state) {
                 return &c->startup_cpuset_cpus;
-        else
-                return &c->cpuset_cpus;
+                        return &c->cpuset_cpus;
```
indentation?

yuwata commented on src/core/cgroup.c:
```diff
@@ -1054,26 +1054,24 @@ uint64_t cgroup_context_cpu_weight(CGroupContext *c, ManagerState state) {
                 return &c->startup_cpuset_mems;
-        else
-                return &c->cpuset_mems;
+                        return &c->cpuset_mems;
```
Also here?

yuwata commented on src/core/dbus-execute.c:
```diff
@@ -2908,7 +2917,7 @@ int bus_exec_context_set_transient_property(
                 return 1;
 
-        } else if (streq(name, "SystemCallLog")) {
+        } if (streq(name, "SystemCallLog")) {
```
Missing line break.
Also, please drop the empty line before `}`

yuwata commented on src/core/dbus-execute.c:
```diff
@@ -3101,7 +3112,9 @@ int bus_exec_context_set_transient_property(
 
                 return 1;
 
```
Please drop the empty line.

yuwata commented on src/core/dbus-execute.c:
```diff
@@ -3328,10 +3348,11 @@ int bus_exec_context_set_transient_property(
 
                 return 1;
 
```
Also here.

yuwata commented on src/core/dbus-execute.c:
```diff
@@ -3969,7 +4003,9 @@ int bus_exec_context_set_transient_property(
 
                 return 1;
 
```
Also here.

yuwata commented on src/core/job.c:
```diff
@@ -254,19 +254,20 @@ Job* job_install(Job *j) {
+                        job_merge_into_installed(uj, j);
+                        log_unit_debug(uj->unit,
+                                        "Merged into running job, re-running: %s/%s as %"PRIu32,
+                                        uj->unit->id, job_type_to_string(uj->type), uj->id);
```
Indentation.

yuwata commented on src/core/unit-serialize.c:
```diff
@@ -233,66 +233,74 @@ int unit_deserialize_state(Unit *u, FILE *f, FDSet *fds) {
                                 log_unit_warning(u, "Update from too old systemd versions are unsupported, cannot deserialize job: %s", v);
                         continue;
-                } else if (streq(l, "state-change-timestamp")) {
+                } if (streq(l, "state-change-timestamp")) {
```
Missing line break.

yuwata commented on src/core/unit-serialize.c:
```diff
@@ -233,66 +233,74 @@ int unit_deserialize_state(Unit *u, FILE *f, FDSet *fds) {
+                if (streq(l, "auto-start-stop-ratelimit")) {
                         deserialize_ratelimit(&u->auto_start_stop_ratelimit, l, v);
                         continue;
 
```
Please drop the empty line.

yuwata commented on src/core/unit-serialize.c:
```diff
@@ -302,7 +310,9 @@ int unit_deserialize_state(Unit *u, FILE *f, FDSet *fds) {
                                 unit_ref_uid_gid(u, uid, GID_INVALID);
                         continue;
 
```
Please drop the empty line.

yuwata commented on src/journal-remote/journal-remote.c:
```diff
@@ -395,20 +395,25 @@ int journal_remote_handle_raw_source(
                 return 1;
-        } else if (r == -EAGAIN) {
+        }
+        if (r == -EAGAIN) {
```
Unnecessary `{}`

yuwata commented on src/libsystemd/sd-bus/sd-bus.c:
```diff
@@ -89,7 +89,7 @@ static sd_bus** bus_choose_default(int (**bus_open)(sd_bus **)) {
                                 *bus_open = sd_bus_open_system;
                         return &default_system_bus;
-                } else if (STR_IN_SET(e, "user", "session")) {
+                } if (STR_IN_SET(e, "user", "session")) {
```
Missing line break.

yuwata commented on src/libsystemd/sd-bus/sd-bus.c:
```diff
@@ -112,11 +112,10 @@ static sd_bus** bus_choose_default(int (**bus_open)(sd_bus **)) {
                 return &default_user_bus;
-        } else {
-                if (bus_open)
+        }                 if (bus_open)
```
Missing line break.

yuwata commented on src/libsystemd/sd-bus/sd-bus.c:
```diff
@@ -112,11 +112,10 @@ static sd_bus** bus_choose_default(int (**bus_open)(sd_bus **)) {
                         *bus_open = sd_bus_open_system;
                 return &default_system_bus;
-        }
+
```
Spurious indentation and unnecessary empty line.

yuwata commented on src/libsystemd/sd-bus/sd-bus.c:
```diff
@@ -2462,14 +2459,16 @@ _public_ int sd_bus_call(
+                        }
+
+                        if (BUS_MESSAGE_COOKIE(incoming) == cookie &&
                                    bus->unique_name &&
```
indentation.

yuwata commented on src/network/networkd-setlink.c:
```diff
@@ -946,24 +946,28 @@ int link_request_to_set_master(Link *link) {
                 return link_request_set_link(link, REQUEST_TYPE_SET_LINK_MASTER,
                                              link_set_master_handler,
                                              NULL);
 
```
Please drop the empty line.

yuwata commented on src/network/networkd-setlink.c:
```diff
@@ -946,24 +946,28 @@ int link_request_to_set_master(Link *link) {
                 return link_request_set_link(link, REQUEST_TYPE_SET_LINK_MASTER,
                                              link_unset_master_handler,
                                              NULL);
 
```
Also here.

yuwata commented on src/nsresourced/nsresourcework.c:
```diff
@@ -1903,7 +1903,9 @@ static int vl_method_add_netif_to_user_namespace(sd_varlink *link, sd_json_varia
                                 SD_JSON_BUILD_PAIR_STRING("hostInterfaceName", ifname_host),
                                 SD_JSON_BUILD_PAIR_STRING("namespaceInterfaceName", ifname_namespace));
 
```
Please drop the empty line.

yuwata commented on src/resolve/resolved-dns-scope.c:
```diff
@@ -1781,16 +1785,18 @@ bool dns_scope_is_default_route(DnsScope *scope) {
                  * volunteer as default route. */
                 return !dns_scope_has_route_only_domains(scope);
 
```
Please drop the empty line.

yuwata commented on src/shared/ask-password-api.c:
```diff
@@ -431,7 +431,9 @@ int ask_password_plymouth(
                         /* No password, because UI not shown */
                         return -ENOENT;
 
```
Also here.

yuwata commented on src/shared/ask-password-api.c:
```diff
@@ -457,9 +459,10 @@ int ask_password_plymouth(
                         *ret = TAKE_PTR(l);
                         return 0;
 
```
Also here.

yuwata commented on src/shared/battery-util.c:
```diff
@@ -169,13 +171,13 @@ int on_ac_power(void) {
                 log_debug("Found at least one online non-battery power supply, system is running on AC.");
                 return true;
-        } else if (found_discharging_battery) {
+        } if (found_discharging_battery) {
```
Missing line break.

yuwata commented on src/shared/condition.c:
```diff
@@ -604,9 +604,10 @@ static int condition_test_firmware(Condition *c, char **env) {
-                } else
-                        return true;
-        } else if ((arg = startswith(c->parameter, "device-tree-compatible("))) {
+                }                         return true;
```
Missing line break after `}`

yuwata commented on src/shared/efi-api.c:
```diff
@@ -96,7 +96,9 @@ static int get_os_indications(uint64_t *ret) {
                 *ret = 0;
                 return 0;
 
```
Please drop the empty line.

yuwata commented on src/shared/vpick.c:
```diff
@@ -315,7 +315,8 @@ static int make_choice(
                                 flags & ~PICK_RESOLVE,
                                 ret);
 
```
Please drop the empty line.

yuwata commented on src/systemctl/systemctl-set-default.c:
```diff
@@ -31,11 +31,11 @@ static int parse_proc_cmdline_item(const char *key, const char *value, void *dat
 
                 return free_and_strdup_warn(ret, key);
 
```
Also here.

yuwata commented on src/systemctl/systemctl-show.c:
```diff
@@ -2363,7 +2363,9 @@ static int show_one(
 
                 return EXIT_PROGRAM_RUNNING_OR_SERVICE_OK;
 
```
Also here.

yuwata commented on src/sysupdate/sysupdate.c:
```diff
@@ -1166,48 +1165,47 @@ static int verb_list(int argc, char **argv, void *userdata) {
-                        return log_error_errno(r, "Failed to create JSON: %m");
+        r = sd_json_buildo(&json, SD_JSON_BUILD_PAIR_STRING("current", current),
+                                        SD_JSON_BUILD_PAIR_STRV("all", versions),
+                                        SD_JSON_BUILD_PAIR_STRV("appstreamUrls", appstream_urls));
```
indentation?

yuwata commented on src/boot/efi-string-table.h:
```diff
@@ -31,3 +31,13 @@
+
+#define DECLARE_STRING_TABLE_LOOKUP(name, type) \
+        DECLARE_STRING_TABLE_LOOKUP_TO_STRING(name, type) \
+        DECLARE_STRING_TABLE_LOOKUP_FROM_STRING(name, type)
```
Hmm, these should be moved to fundamental-string-table.h?
Then, maybe better to include it in basic-forward.h?

yuwata commented on src/basic/uid-classification.c:
```diff
@@ -1,12 +1,15 @@
 /* SPDX-License-Identifier: LGPL-2.1-or-later */
 
+#if ENABLE_COMPAT_MUTABLE_UID_BOUNDARIES
```
Hmm, if we do like that, we may easily break build, as IIRC this is never tested in build test GH action.

Please enable the option in one build test.

daandemeyer committed:
sd-journal: Remove const from function parameter

boot_id is already passed by value, and hence copied.
Since we don't apply const to function parameters
that are copied anywhere else, let's drop the const
here as well for consistency.

daandemeyer committed:
ci: Add coverage for -Dcompat-mutable-uid-boundaries=true

yuwata commented on src/fundamental/string-table-fundamental.h:
```diff
@@ -0,0 +1,21 @@
+
+#include "macro-fundamental.h"
+
+#define DECLARE_STRING_TABLE_LOOKUP_TO_STRING(name, type) \
```
Uh, this is added in a wrong commit...

daandemeyer replied:
oops

yuwata commented on src/systemd/sd-json.h:
```diff
@@ -335,7 +335,7 @@ int sd_json_variant_strv(sd_json_variant *v, char ***ret);
 int sd_json_variant_unhex(sd_json_variant *v, void **ret, size_t *ret_size);
 
-const char* sd_json_variant_type_to_string(sd_json_variant_type_t t);
+const char* sd_json_variant_type_to_string(sd_json_variant_type_t i);
```
How about to also introduce a macro in _sd_common.h, like `_SD_DECLARE_TRIVIAL_REF_FUNC()` you added?
Though, not strong request, as it will be only used here at least now.

daandemeyer replied:
I would leave it until we have more use cases

yuwata replied:
No problem.

keszybz commented on src/basic/log-context.h:
```diff
@@ -57,8 +57,7 @@ LogContext* log_context_new_iov(struct iovec *input_iovec, size_t n_input_iovec,
 
-LogContext *log_context_ref(LogContext *c);
-LogContext *log_context_unref(LogContext *c);
+DECLARE_TRIVIAL_REF_UNREF_FUNC(LogContext, log_context);
```
~~I don't understand. In other commits the change was made to use short names like `const char *s` and such. Here, the change is 180° opposite. ??~~
Ah, OK, the variable is `*p`. I mislooked.

keszybz commented on src/basic/string-table.h:
```diff
@@ -72,17 +70,16 @@ ssize_t string_table_lookup_from_string_fallback(const char * const *table, size
                                 continue;                               \
-                        fputs_unlocked(_t, stdout);                     \
-                        fputc_unlocked('\n', stdout);                   \
+                        dump_string_table_item(_t);                     \
```
Dropping of the locking looks reasonable. But why do we need `dump_string_table_item`? Just to avoid the include of stdio.h here? Is it really worth the trouble?

daandemeyer replied:
Not anymore now that we have string-table-fundamental.h, I'll drop the commit.

daandemeyer committed:
tree-wide: Drop NOLINTNEXTLINE comments

Not needed anymore after recent refactoring.
